Ssuto 70 refactor user profile form to be reactive

mvance-smoothstack opened the pull request:
Pulled in ecc81a2 and fixed the form to properly pass values through as displayed to the user, and also added phone number display.

mvance-smoothstack committed:
cleaned up user profile elements

mvance-smoothstack committed:
implemented user profile update functionality

mvance-smoothstack committed:
user profile form now has styling and validation

mvance-smoothstack committed:
last commit for big merge

mvance-smoothstack committed:
Merge branch 'dev' of https://github.com/jms-smoothstack-utopia/UtopiaUserPortal into SSUTO-7-RegisteredUserUpdatesProfileInformation

mvance-smoothstack committed:
added functionality for notification settings

mvance-smoothstack committed:
added tests for user profile, service

mvance-smoothstack committed:
profile form now reactive, shows loyalty points

mvance-smoothstack committed:
wrote more tests and changed user ID type

sgerkin-ss committed:
Fix login endpoint

mvance-smoothstack committed:
hotfix to pass values through to display

and added user's phone number onto the form

mvance-smoothstack committed:
updated auth tests to use new login URL

(see ecc81a2)

jordanDSmoothStack reviewed (commented):
LGTM; just took a brief look though

sgerkin-ss committed:
Fix issue with creating a profile not waiting for login success.

When creating an account, a user is to be redirected to their profile;
however, this requires the user to be logged in. This was being performed,
but the result of logging in was not received before issuing a request to
get profile information first. This has been corrected to await the login
response before navigating to the user profile.

sgerkin-ss committed:
Fix autologin

mvance-smoothstack committed:
fixes for user profile

user service now uses logger instead of console.log
user service update method now lets the component listen to callbacks
profile component now uses the shared alert modal for update results
profile compoent formats the phone number nicely
and reformats it for the backend
loyalty points no longer in a form control, just static
